synapse/replication/http/register.py

```diff
@@ -75,6 +75,8 @@ def _serialize_payload(
     async def _handle_request(self, request, user_id):
         content = parse_json_object_from_request(request)
 
+        await self.registration_handler.check_registration_ratelimit(content["address"])
```

Comment:
```suggestion
        self.registration_handler.check_registration_ratelimit(content["address"])
```

This doesn't return a `Deferred` so can't be awaited on